lib/identity_job_log_subscriber.rb

```diff
@@ -109,63 +83,30 @@ def perform(event)
   def enqueue_retry(event)
     job = event.payload[:job]
     ex = event.payload[:error]
-    wait_seconds = event.payload[:wait]
 
-    json = default_attributes(event, job).merge(
-      wait_ms: wait_seconds.to_i.in_milliseconds,
-    )
+    wait_seconds = event.payload[:wait]
+    wait_ms = wait_seconds.to_i.in_milliseconds
 
     if ex
-      if should_error?(job, ex)
-        json[:exception_class] = ex.class.name
-        error(json.to_json)
-      else
-        json[:exception_class_warn] = ex.class.name
-        info(json.to_json)
-      end
+      error_or_warn(event: event, extra_attributes: { wait_ms: wait_ms })
+    else
+      default_attributes(event, job).merge(
+        wait_ms: wait_ms,
+      )
     end
-
-    json
   end
 
   def retry_stopped(event)
     job = event.payload[:job]
-    ex = event.payload[:error]
-
-    if should_error?(job, ex)
-      json = default_attributes(event, job).merge(
-        exception_class: ex.class.name,
-        attempts: job.executions,
-      )
 
-      error(json.to_json)
-    else
-      json = default_attributes(event, job).merge(
-        exception_class_warn: ex.class.name,
-        attempts: job.executions,
-      )
-
-      warn(json.to_json)
-    end
+    error_or_warn(
+      event: event,
+      extra_attributes: { attempts: job.executions },
+    )
   end
 
   def discard(event)
-    job = event.payload[:job]
-    ex = event.payload[:error]
-
-    if should_error?(job, ex)
-      json = default_attributes(event, job).merge(
-        exception_class: job.class,
-      )
-
-      error(json.to_json)
-    else
-      json = default_attributes(event, job).merge(
-        exception_class_warn: job.class,
-      )
```

Comment:
I'm 90% certain `job.class` was a bug, it should have been `ex.class.name`, and the spec changes seemed to confirm this